File: flutter_version/lib/src/features/news/presentation/bloc/news_state.dart
import '../../domain/entities/article.dart';

abstract class NewsState {}

class NewsInitial extends NewsState {}

class NewsLoading extends NewsState {}

class NewsLoaded extends NewsState {
  final List<Article> articles;
  final bool hasReachedMax;
  final bool isLoadingMore;

  NewsLoaded(
    this.articles, {
    this.hasReachedMax = false,
    this.isLoadingMore = false,
  });
}

class NewsError extends NewsState {
  final String message;
  NewsError(this.message);
}

File: compose_version/app/src/main/java/com/example/news_app/presentation/viewmodel/NewsState.kt
package com.example.news_app.presentation.viewmodel

import com.example.news_app.domain.entities.Article

sealed class NewsState {
    object Initial : NewsState()
    object Loading : NewsState()
    data class Loaded(
        val articles: List<Article>,
        val hasReachedMax: Boolean = false,
        val isLoadingMore: Boolean = false,
    ) : NewsState()
    data class Error(val message: String) : NewsState()
}
